### ISSUE TITLE
mj-wrapper Adds unwanted padding on the bottom on Outlook

### ISSUE THREAD

**Issue by janus-reith** [mjmlio/mjml#713]
Outllook adds ~20px of padding under the last Element of the wrapper.
Example:
```
<mjml>
  <mj-body>
    <mj-container>
      <mj-wrapper background-url="http://phplist.cds-service.com/images/img/2017/Redesign/shadow_over600.jpg" full-width="full-width" padding="0" background-repeat="repeat-y">
        <mj-section background-color="#75b842" padding="0" padding-bottom="18" padding-top="18">
          <mj-column>
            <mj-text align="center" color="#FFFFFF" padding="0" alt="">Example 1
            </mj-text>
          </mj-column>
          <mj-column>
            <mj-text align="center" color="#FFFFFF" padding="0" alt="">Example 2
            </mj-text>
          </mj-column>
          <mj-column>
            <mj-text align="center" color="#FFFFFF" padding="0" alt="">Example 3
            </mj-text>
          </mj-column>
        </mj-section>
      </mj-wrapper>
    </mj-container>
  </mj-body>
</mjml>
```

Expected Output:
![index-desktop](https://user-images.githubusercontent.com/1831220/27080175-0a7d717e-503b-11e7-8714-c7c307c1353a.png)

Outlook 2016(probably older versions aswell, didn't test yet):
![image](https://user-images.githubusercontent.com/1831220/27080240-4caac812-503b-11e7-8f0e-17168325d4e2.png)

**Notice:** There're additional tweaks needed for the generated html to make the bg show properly at all, as mj-wrapper has some other Outlook bugs:

`<v:rect xmlns:v="urn:schemas-microsoft-com:vml" fill="true" stroke="false" style="width:600px;">`
has to be changed to `style="width:640px;"`
Why? - Because this is the width of my border shadow, which would else be cropped at 600px width, although the wrapper is set to full-width.
I would like to keep that dynamic, but `mso-width-percent:1000` somehow messed up the layout aswell.

Also, every instance of `type="tile"` has to be replaced with `type="frame"`, else the scaling is weird.

**Comment by iRyusa** [comment]
Hi @janus-reith glad to hear from you again !

Full-width section are not yet supported in Outlook desktop clients, so that's why you have a 600px background. However, I think there's an open issue from you about this behaviour from you in #116 and in #128.

You had a PR about tile / frame here(#165), I don't know if it should be a "expert" attribute like outlook-fill-type or something..

**Comment by janus-reith** [comment]
Yes, most of these issues except the mj-wrapper 600px width are the same that I also have with section backgrounds.
Before I used 600px sections with a 600px background shadow and padding left and right, now I'm trying to use the whole 600px and wrap the 640px background image around it.

I never finished the type="tile"<=>frame pr, as I didnt know how to not break it for people that have horizontal tiled images and need type="tile" - The idea of an specific "expert" attribute seems to be a viable solution.

**Comment by janus-reith** [comment]
Still searching for a solution to remove the unwanted padding.
Manually editing the output html would also be okay as a Start.

**Comment by janus-reith** [comment]
Just tested, It occurs on all Outlook versions from 2007-2016, except the Mac versions.

**Comment by janus-reith** [comment]
This is driving me crazy - I just tried to manully add the backgrounds.cm code to my html output without setting a bg image in the mjml code - The issue is the same and is somehow created by the <v:rect><v:fill><v:textbox> part.

Someone here seems to have a similar issue: https://litmus.com/community/discussions/538-vml-outlook-07-10-13-unwanted-20px-padding-at-the-bottom

They solved it by adding another `<div style="font-size: 0;">` inside of the first div that comes after the  vml part.

Sadly I couldn't reproduce that solution, as the first thing that comes after the vml in our mjml code is not a `<div>` but a `<table>`. Still tried different combinations without success.

**Comment by ogonkov** [comment]
Hack with adding `<p style="margin:0;mso-hide:all"><o:p xmlns:o="urn:schemas-microsoft-com:office:office">&nbsp;</o:p></p>` right before `</v:textbox>` works for me.

Should i make PR?

**Comment by iRyusa** [comment]
This has been released in MJML 3.3.5 now so I can close this !
